Backend/app/kafka-consumers/src/Consumer/CustomerStats/Processor.hs

```diff
@@ -0,0 +1,78 @@
+import "dynamic-offer-driver-app" Domain.Types.Ride as DDR
+import Environment
+-- import qualified Kernel.Storage.Esqueleto as Esq
+
+import Kernel.Beam.Functions
+import Kernel.Prelude
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import "sessionizer-metrics" Lib.SessionizerMetrics.Types.Event as E
+import "rider-app" SharedLogic.Person as SP
+import "rider-app" Storage.CachedQueries.Merchant as CQMerchant
+import "rider-app" Storage.Queries.BookingCancellationReason as QBCR
+import "rider-app" Storage.Queries.Person.PersonStats as QP
+import "rider-app" Tools.Error
+import "dynamic-offer-driver-app" Tools.Event as TE
+
+updateCustomerStats :: E.Event TE.Payload -> Text -> Flow ()
+updateCustomerStats event _ = do
+  whenJust event.personId $ \pId -> do
+    when (event.service == RIDER_APP) do
+      let personId = Id pId
+          merchantId = Id event.merchantId
+      -- personStats <- Esq.runInReplica $ QP.findByPersonId personId >>= fromMaybeM (PersonStatsNotFound personId.getId)
+      personStats <- runInReplica $ QP.findByPersonId personId >>= fromMaybeM (PersonStatsNotFound personId.getId)
+      when (isNotBackfilled personStats) do
+        personStatsValues <- SP.backfillPersonStats personId merchantId
+        -- Esq.runNoTransaction $ QP.incrementOrSetPersonStats personStatsValues
+        QP.incrementOrSetPersonStats personStatsValues
+      whenJust (event.payload) $ \payload -> do
+        case event.eventType of
+          RideEnded -> do
+            case payload.rs of
+              DDR.COMPLETED -> do
+                let createdAt = payload.cAt
+                merchant <- CQMerchant.findById merchantId >>= fromMaybeM (MerchantDoesNotExist merchantId.getId)
+                let minuteDiffFromUTC = (merchant.timeDiffFromUtc.getSeconds) `div` 60
+                unless (isNotBackfilled personStats) do
+                  -- Esq.runNoTransaction $ do
+                  when (SP.isWeekend createdAt minuteDiffFromUTC) do QP.incrementWeekendRidesCount personId
+                  unless (SP.isWeekend createdAt minuteDiffFromUTC) do QP.incrementWeekdayRidesCount personId
+                  when (SP.checkMorningPeakInWeekday createdAt minuteDiffFromUTC) do QP.incrementMorningPeakRidesCount personId
+                  when (SP.checkEveningPeakInWeekday createdAt minuteDiffFromUTC) do QP.incrementEveningPeakRidesCount personId
+                  when (SP.checkWeekendPeak createdAt minuteDiffFromUTC) do QP.incrementWeekendPeakRidesCount personId
+                  when (not (SP.checkEveningPeakInWeekday createdAt minuteDiffFromUTC) && not (SP.checkMorningPeakInWeekday createdAt minuteDiffFromUTC) && not (SP.checkWeekendPeak createdAt minuteDiffFromUTC)) do QP.incrementOffpeakRidesCount personId
+                  QP.incrementCompletedRidesCount personId
+              _ -> pure ()
+          BookingCancelled -> do
+            unless (isNotBackfilled personStats) do
```

Comment:
This condition is in both cases, so can be present at the top

Reply:
done